noir/noir-repo/acvm-repo/bn254_blackbox_solver/src/lib.rs

```diff
@@ -48,6 +48,15 @@ impl Default for Bn254BlackBoxSolver {
 }
 
 impl BlackBoxFunctionSolver for Bn254BlackBoxSolver {
+    fn aes128_encrypt(
+        &self,
+        inputs: &[u8],
+        iv: [u8; 16],
+        key: [u8; 16],
+    ) -> Result<Vec<u8>, BlackBoxResolutionError> {
+        aes128_encrypt(inputs, iv, key)
+    }
```

Comment:
This isn't specific to the bn254 curve so it should be in the generic blackbox solver crate.

Reply:
Not sure I understand what you mean, it is implemented in the generic blackbox solver crate, I actually pull it from there just to fulfill the trait?

Reply:
This trait is specifically for black box functions which are defined based on the underlying curve, e.g. fixed_base_scalar_mul acts on grumpkin because the ACVM is currently using the bn254 field. The idea being that if you swap out the field then you also swap out this crate as you need solvers for your new curve.

AES doesn't depend on the curve so it shouldn't be included on this trait, similarly to XOR, keccak256, ECDSA signature verification, etc.

Reply:
Understood and fixed. Naming is a little bit deceptive though!